report.sh

```diff
@@ -1,10 +1,10 @@
 #!/bin/bash
 # ------------------------------------------------------------------------ #
-# Script Name:   report.sh 
+# Script Name:   report.sh
 # Description:   An interactive tool to check the deatils of your system.
 # Written by:    Ranabir Chakraborty
 # ------------------------------------------------------------------------ #
-
+set -euo pipefail
 
 function menuprincipal () {
```

Comment:
This function is HUGE! You should put all the fonction it contains above it to make it more readable.